src/lib/gallery.ts

```diff
@@ -0,0 +1,192 @@
+import { get as getStore } from 'svelte/store'
+import * as wn from 'webnative'
+import { filesystemStore, galleryStore } from '../stores'
+import { convertUint8ToString } from '$lib/common/utils'
+import { addNotification } from '$lib/notifications'
+
+export enum AREAS {
+  PUBLIC = 'Public',
+  PRIVATE = 'Private',
+}
+
+export type Image = {
+  cid: string
+  ctime: number
+  name: string
+  private: boolean
+  size: number
+  src: string
+}
+
+export type Gallery = {
+  publicImages: Image[] | null
+  privateImages: Image[] | null
+  selectedArea: AREAS
+  loading: boolean
+}
+
+export const GALLERY_DIRS = {
+  [AREAS.PUBLIC]: ['public', 'gallery'],
+  [AREAS.PRIVATE]: ['private', 'gallery'],
+}
+const FILE_SIZE_LIMIT = 5
+
+/**
+ * Get images from the user's WNFS and construct the `src` value for the images
+ */
+export const getImagesFromWNFS: () => Promise<void> = async () => {
+  try {
+    // Set loading: true on the galleryStore
+    galleryStore.update((store) => ({ ...store, loading: true }))
+
+    const { selectedArea } = getStore(galleryStore)
+    const isPrivate = selectedArea === AREAS.PRIVATE
+    const fs = getStore(filesystemStore)
+
+    // Set path to either private or public gallery dir
+    const path = wn.path.directory(...GALLERY_DIRS[selectedArea])
+
+    // Get list of links for files in the gallery dir
+    const links = await fs.ls(path)
+
+    const images = await Promise.all(
+      Object.entries(links).map(async ([name]) => {
+        const file = await fs.get(
+          wn.path.file(...GALLERY_DIRS[selectedArea], `${name}`)
+        )
+
+        // The CID for private files is currently located in `file.header.content`,
+        // whereas the CID for public files is located at `file.cid`
+        const cid = isPrivate ? file.header.content.toString() : file.cid.toString()
```

Comment:
I'm seeing some type errors here and in a couple of other places below:

```
Property 'content' does not exist on type 'PuttableUnixTree | File'.
  Property 'content' does not exist on type 'PuttableUnixTree'.`
```

These are the types returned by `fs.get`, but they don't appear to have some of the properties we use like `header` and `cid`. These properties instead appear on the `PrivateFile` and `PublicFile` class interfaces in webnative.

I wonder if there is a way we can type `file` better to avoid the type errors. @matheus23 do you have any ideas for how we might do this?

Reply:
Yeah, we should fix that. At this point I'd consider `.header` part of the public api.

Reply:
Actually, we should make note of these use cases here and improve the webnative api to specifically support getting a "hard link" for stuff. The `.content` field is encoding-specific and will change in newer versions of WNFS.

What's the use case for this cid here exactly? Where is it used?

Reply:
The `cid` is used for a couple of things. It's an identifier for the images like here:

https://github.com/fission-codes/webnative-app-template/blob/140f8a6187ce219df04caa1f6baf8b148b1769f9/src/components/gallery/imageGallery/ImageModal.svelte#L95

But more importantly, it used for creating direct links to the images on a gateway here:

https://github.com/fission-codes/webnative-app-template/blob/140f8a6187ce219df04caa1f6baf8b148b1769f9/src/components/gallery/imageGallery/ImageModal.svelte#L140

Reply:
Okay. Well, that direct link will not really be useful to a user when generated for a private file. Specifically, the `/userland` path can't be resolved. And the `https://ipfs.io/ipfs/${image.cid}` link will just return you a bag of encrypted bytes, so likely not useful to the user.

In the future there won't be a "single CID" for a private file, so that API will break. The file will be broken up into lots of pieces and can't be addressed using a single CID anymore for privacy.

Reply:
The thought here was that a developer using the template would open the private file and realize that it was encrypted, but you're right that it wouldn't be much use to a user. 

Considering that the API will change, it might be better to remove the link and comment or document in the README why a link is not implemented for private files.